includes/Core/Storage/Data_Encryption.php

```diff
@@ -81,21 +83,36 @@ public function encrypt( $value ) {
 	 *
 	 * @param string $raw_value Value to decrypt.
 	 * @return string|bool Decrypted value, or false on failure.
+	 * @throws \Exception If the value fails to decode.
 	 */
 	public function decrypt( $raw_value ) {
-		if ( ! extension_loaded( 'openssl' ) ) {
+		if ( ! extension_loaded( 'openssl' ) || ! is_string( $raw_value ) ) {
 			return $raw_value;
 		}
 
-		$raw_value = base64_decode( $raw_value, true );
+		$decoded_value = base64_decode( $raw_value, true );
+
+		if ( false === $decoded_value ) {
+			if ( 'development' === Build_Mode::get_mode() ) {
```

Comment:
I don't see this actually being used anywhere else, I don't even think we need an Exception per-se, I was thinking more something like `error_log()`.

Reply:
@tofumatt I initially added that one, but due to the phpcs warnings, I switched to throw error, and due to nature of fatal error I can throw this only in development env, vs trying to add phpcs ignore rule. 
It seemed better to throw error and do it only in development, so in case it happens, it won't be missed. What do you think?